### PR TITLE
fix(Script/Quest): Doing Your Duty

### PR DESCRIPTION
**Pull request by Vanheden**
Roleplay & Script Fix For Doing Your Duty Quest

Credits to Kitzunu for the Script & Bench for the Sniff!

## Changes Proposed:
None that I know of.

## Issues Addressed:
Fixes the Roleplay after the Quest has been completed.
Fixes the Amberhouse Script during the Quest.

Closes: https://github.com/azerothcore/azerothcore-wotlk/issues/16347

## SOURCE:
Youtube.
Sniff.
Wotlk Classic.
WoW Retail.

## Tests Performed:
It's tested on Azerothcore with a video here.
[Video](https://www.youtube.com/watch?v=piTHWbaIgm0&ab_channel=Roddan)

## How to Test the Changes:

1. .q add 12227
2. Run to the Amberhouse and use it.
3. Watch the Roleplay after the quest has been turned in.

## Known Issues and TODO List:
Spell Dust Field (48329) currently not working when inside the Amberhouse.

### CODE REVIEW

File: data/sql/updates/pending_db_world/Roleplay.sql [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+DELETE FROM `waypoints` WHERE (`entry` = 2707100) AND (`pointid` = 4);
+DELETE FROM `waypoints` WHERE (`entry` = 2707100) AND (`pointid` = 5);
+DELETE FROM `waypoints` WHERE (`entry` = 2707100) AND (`pointid` = 6);
+DELETE FROM `waypoints` WHERE (`entry` = 2707100) AND (`pointid` = 7);
+DELETE FROM `waypoints` WHERE (`entry` = 2707100) AND (`pointid` = 8);
+DELETE FROM `waypoints` WHERE (`entry` = 2707100) AND (`pointid` = 9);
+DELETE FROM `waypoints` WHERE (`entry` = 2707100) AND (`pointid` = 10);
+DELETE FROM `waypoints` WHERE (`entry` = 2707100) AND (`pointid` = 11);
+DELETE FROM `waypoints` WHERE (`entry` = 2707100) AND (`pointid` = 12);
+DELETE FROM `waypoints` WHERE (`entry` = 2707100) AND (`pointid` = 13);
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/Roleplay.sql:33]: DELETE FROM `waypoints` WHERE (`entry` = 2707100) AND (`pointid` BETWEEN 1 AND 13);

File: data/sql/updates/pending_db_world/Roleplay.sql [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+DELETE FROM `creature_text` WHERE `CreatureID` = 27277;
+INSERT INTO `creature_text` (`CreatureID`, `GroupID`, `ID`, `Text`, `Type`, `Language`, `Probability`, `Emote`, `Duration`, `Sound`, `BroadcastTextId`, `TextRange`, `comment`) VALUES
+(27277, 0, 0, 'Alright Ben. We\'ve retrieved the amberseeds again. You know the drill.', 12, 0, 100, 0, 0, 0, 26350, 0, 'Master Woodsman Anderhol // Master Woodsman Anderhol');
+
+DELETE FROM `creature_text` WHERE `CreatureID` = 27071;
+INSERT INTO `creature_text` (`CreatureID`, `GroupID`, `ID`, `Text`, `Type`, `Language`, `Probability`, `Emote`, `Duration`, `Sound`, `BroadcastTextId`, `TextRange`, `comment`) VALUES
+(27071, 0, 0, 'I know, I know. Back in the bucket....', 12, 0, 100, 0, 0, 0, 26351, 0, 'Benjamin Jacobs // Benjamin Jacobs');
+
+DELETE FROM `waypoints` WHERE (`entry` = 27071) AND (`pointid` = 1);
+DELETE FROM `waypoints` WHERE (`entry` = 27071) AND (`pointid` = 2);
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/Roleplay.sql:20]: DELETE FROM `waypoints` WHERE (`entry` = 27071) AND (`pointid` IN (1, 2));

File: data/sql/updates/pending_db_world/Roleplay.sql [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+(@ENTRY, 0, 4, 0, 59, 0, 100, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'On timed event timedEvent[1] triggered - Self: Talk I know, I know. Back in the bucket.... (0) to invoker'),
+(@ENTRY, 0, 5, 0, 59, 0, 100, 0, 2, 0, 0, 0, 53, 0, 2707100, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'On timed event timedEvent[2] triggered - Self: Start path #2707100, walk, do not repeat, Passive'),
+(@ENTRY, 0, 6, 7, 59, 0, 100, 0, 3, 0, 0, 0, 90, 8, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'On timed event timedEvent[3] triggered - Self: Set stand state to KNEEL'),
+(@ENTRY, 0, 7, 0, 61, 0, 100, 0, 0, 0, 0, 0, 41, 6000, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'On timed event timedEvent[3] triggered - Self: Despawn in 6 s respawn in 1 seconds'),
+(@ENTRY, 0, 8, 0, 11, 0, 100, 0, 0, 0, 0, 0, 91, 8, 0, 0, 0, 0, 0, 10, 113907, 27071, 0, 0, 0, 0, 0, 'On respawn - Creature Benjamin Jacobs (27071) with guid 113907 (fetching): Remove stand state KNEEL');
+
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId` = 22 AND `SourceEntry` = 27071 AND `SourceId` = 0;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/Roleplay.sql:66]: add a newline

File: data/sql/updates/pending_db_world/amberhouse.sql [hunk 4]
```diff
@@ -0,0 +1,12 @@
+UPDATE `creature_template` SET `scriptname` = '' WHERE `entry` = 27326;
+
+DELETE FROM `spell_target_position` WHERE `ID` = 48324 AND `EffectIndex` = 0;
+INSERT INTO `spell_target_position` (`ID`, `EffectIndex`, `MapID`, `PositionX`, `PositionY`, `PositionZ`, `Orientation`, `VerifiedBuild`) VALUES
+(48324, 0, 571, 3454.11, -2802.37, 202.14, 0, 34149345);
+
+DELETE FROM `spell_script_names` WHERE `spell_id` IN (48382, 53017);
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
+(48382, 'spell_q12227_outhouse_groans'),
+(53017, 'spell_q12227_indisposed_i');
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/amberhouse.sql:10]:
```suggestion
DELETE FROM `spell_script_names` WHERE `spell_id` IN (48382);
INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
(48382, 'spell_q12227_outhouse_groans');
```

File: src/server/scripts/Northrend/zone_grizzly_hills.cpp [hunk 5]
```diff
@@ -319,71 +319,55 @@ class npc_ravenous_worg : public CreatureScript
 };
 
 /*######
-## Quest 12227: Doing Your Duty
-######*/
++## Quest 12227: Doing Your Duty
++######*/
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/zone_grizzly_hills.cpp:323]:
```suggestion
## Quest 12227: Doing Your Duty
######*/
```

File: src/server/scripts/Northrend/zone_grizzly_hills.cpp [hunk 6]
```diff
@@ -319,71 +319,55 @@ class npc_ravenous_worg : public CreatureScript
...
 {
     // Sound
     SOUND_FEMALE                    = 12671,
     SOUND_MALE                      = 12670,
     // Spell
-    SPELL_OUTHOUSE_GROANS           = 48382,
     SPELL_CAMERA_SHAKE              = 47533,
-    SPELL_DUST_FIELD                = 48329
+    SPELL_DUST_FIELD                = 48329,
+    SPELL_CREATE_AMBERSEEDS         = 48330,
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/zone_grizzly_hills.cpp:333]:
```suggestion
```

File: src/server/scripts/Northrend/zone_grizzly_hills.cpp [hunk 7]
```diff
@@ -1314,4 +1297,6 @@ void AddSC_grizzly_hills()
     new spell_warhead_detonate();
     new spell_vehicle_warhead_fuse();
     new spell_warhead_fuse();
+    RegisterSpellScript(spell_q12227_outhouse_groans);
+    RegisterSpellScript(spell_q12227_indisposed_i);
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/zone_grizzly_hills.cpp:1301]:
```suggestion
```

File: data/sql/updates/pending_db_world/amberhouse.sql [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+
+DELETE FROM `spell_target_position` WHERE `ID` = 48324 AND `EffectIndex` = 0;
+INSERT INTO `spell_target_position` (`ID`, `EffectIndex`, `MapID`, `PositionX`, `PositionY`, `PositionZ`, `Orientation`, `VerifiedBuild`) VALUES
+(48324, 0, 571, 3454.11, -2802.37, 202.14, 0, 34149345);
+
+DELETE FROM `spell_script_names` WHERE `spell_id` IN (48382);
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
+(48382, 'spell_q12227_outhouse_groans');
+
+DELETE FROM `creature` WHERE  `guid`=113841;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/amberhouse.sql:11]: add a newline

**Review by Kitzunu — COMMENTED** [review]
L#1647
```cpp
bool OnGossipSelect(Player* player, GameObject* go, uint32 /*sender*/, uint32 action) override
```
change to
```cpp
bool OnGossipSelect(Player* player, GameObject* /*go*/, uint32 /*sender*/, uint32 action) override
```

File: src/server/scripts/World/go_scripts.cpp [hunk 9]
```diff
@@ -1163,7 +1163,7 @@ class go_fel_crystalforge : public GameObjectScript
         return true;
     }
 
-    bool OnGossipSelect(Player* player, GameObject* go, uint32 /*sender*/, uint32 action) override
+    bool OnGossipSelect(Player* player, GameObject* /*go*/, uint32 /*sender*/, uint32 action) override
```
Comment (Kitzunu) [inline src/server/scripts/World/go_scripts.cpp:1166]: wrong line and definitely wrong script my man

File: data/sql/updates/pending_db_world/amberhouse.sql [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/amberhouse.sql:11]:
```suggestion
UPDATE `creature_template` SET `flags_extra`=`flags_extra`|128 WHERE `entry` =27326;
```

File: src/server/scripts/Northrend/zone_grizzly_hills.cpp [hunk 11]
```diff
@@ -1314,4 +1296,5 @@ void AddSC_grizzly_hills()
     new spell_warhead_detonate();
     new spell_vehicle_warhead_fuse();
     new spell_warhead_fuse();
+    RegisterSpellScript(spell_q12227_outhouse_groans);
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/zone_grizzly_hills.cpp:1299]:
```suggestion
    RegisterSpellScript(spell_q12227_outhouse_groans);
    RegisterSpellScript(spell_q12227_camera_shake);
```

File: src/server/scripts/Northrend/zone_grizzly_hills.cpp [hunk 12]
```diff
@@ -328,62 +328,45 @@ enum Outhouse
...
-    };
+    }
 
-    CreatureAI* GetAI(Creature* creature) const override
+    void Register() override
     {
-        return new npc_outhouse_bunnyAI(creature);
+        OnEffectHitTarget += SpellEffectFn(spell_q12227_outhouse_groans::HandleScriptEffect, EFFECT_0, SPELL_EFFECT_SCRIPT_EFFECT);
     }
 };
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/zone_grizzly_hills.cpp:372]:
```suggestion
class spell_q12227_camera_shake : public SpellScript
{
    PrepareSpellScript(spell_q12227_camera_shake);

    bool Validate(SpellInfo const* /*spellInfo*/) override
    {
        return ValidateSpellInfo({ SPELL_DUST_FIELD });
    }

    void HandleScriptEffect(SpellEffIndex /*effIndex*/)
    {
        if (Player* player = GetCaster()->ToPlayer())
            if (Creature* target = GetClosestCreatureWithEntry(player, NPC_OUTHOUSE_BUNNY_GRIZZLY, 3.0f)) // hackfix: Outhouse bunny doesnt show in any script. But the visual of Dust Field do not show if cast by the player
                target->CastSpell(target, SPELL_DUST_FIELD, true);
    }

    void Register() override
    {
        OnEffectHitTarget += SpellEffectFn(spell_q12227_camera_shake::HandleScriptEffect, EFFECT_0, SPELL_EFFECT_DUMMY);
    }
};
```

File: src/server/scripts/Northrend/zone_grizzly_hills.cpp [hunk 13]
```diff
@@ -328,62 +328,45 @@ enum Outhouse
     SOUND_FEMALE                    = 12671,
     SOUND_MALE                      = 12670,
     // Spell
-    SPELL_OUTHOUSE_GROANS           = 48382,
     SPELL_CAMERA_SHAKE              = 47533,
-    SPELL_DUST_FIELD                = 48329
+    SPELL_DUST_FIELD                = 48329,
+    // Item
+    ITEM_ANDERHOLS_SLIDER_CIDER     = 37247,
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/zone_grizzly_hills.cpp:334]:
```suggestion
    ITEM_ANDERHOLS_SLIDER_CIDER     = 37247,
    // NPC
    NPC_OUTHOUSE_BUNNY_GRIZZLY      = 27326,
```

File: data/sql/updates/pending_db_world/amberhouse.sql [hunk 14]
```diff
@@ -0,0 +1,12 @@
+UPDATE `creature_template` SET `scriptname` = '' WHERE `entry` = 27326;
+
+DELETE FROM `spell_target_position` WHERE `ID` = 48324 AND `EffectIndex` = 0;
+INSERT INTO `spell_target_position` (`ID`, `EffectIndex`, `MapID`, `PositionX`, `PositionY`, `PositionZ`, `Orientation`, `VerifiedBuild`) VALUES
+(48324, 0, 571, 3454.11, -2802.37, 202.14, 0, 34149345);
+
+DELETE FROM `spell_script_names` WHERE `spell_id` IN (48382);
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
+(48382, 'spell_q12227_outhouse_groans');
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/amberhouse.sql:9]:
```suggestion
DELETE FROM `spell_script_names` WHERE `spell_id` IN (48382, 47533);
INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
(48382, 'spell_q12227_outhouse_groans'),
(47533, 'spell_q12227_camera_shake');
```

File: data/sql/updates/pending_db_world/Roleplay.sql [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+DELETE FROM `creature_text` WHERE `CreatureID` = 27277;
+INSERT INTO `creature_text` (`CreatureID`, `GroupID`, `ID`, `Text`, `Type`, `Language`, `Probability`, `Emote`, `Duration`, `Sound`, `BroadcastTextId`, `TextRange`, `comment`) VALUES
+(27277, 0, 0, 'Alright Ben. We\'ve retrieved the amberseeds again. You know the drill.', 12, 0, 100, 0, 0, 0, 26350, 0, 'Master Woodsman Anderhol // Master Woodsman Anderhol');
+
+DELETE FROM `creature_text` WHERE `CreatureID` = 27071;
+INSERT INTO `creature_text` (`CreatureID`, `GroupID`, `ID`, `Text`, `Type`, `Language`, `Probability`, `Emote`, `Duration`, `Sound`, `BroadcastTextId`, `TextRange`, `comment`) VALUES
+(27071, 0, 0, 'I know, I know. Back in the bucket....', 12, 0, 100, 0, 0, 0, 26351, 0, 'Benjamin Jacobs // Benjamin Jacobs');
+
+DELETE FROM waypoints WHERE (entry = 27071) AND (pointid IN (1, 2));
+DELETE FROM waypoints WHERE (entry = 2707100) AND (pointid BETWEEN 1 AND 13);
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/Roleplay.sql:20]: backticks on table and column names. See https://www.azerothcore.org/wiki/sql-standards#general-standards

File: src/server/scripts/Northrend/zone_grizzly_hills.cpp [hunk 16]
```diff
@@ -328,62 +328,45 @@ enum Outhouse
...
-        void SetData(uint32 Type, uint32 Data) override
+    void HandleScriptEffect(SpellEffIndex /*effIndex*/)
+    {
+        if (Player* player = GetCaster()->ToPlayer())
         {
-            if (Type == 1)
-                _gender = Data;
-        }
+            player->CastSpell(player, SPELL_CAMERA_SHAKE, true);
+            player->CastSpell(player, SPELL_DUST_FIELD, true);
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/zone_grizzly_hills.cpp:351]:
```suggestion
```

**Comment by Kitzunu** [comment]
RP looks good

**Comment by Vanheden** [comment]
> RP looks good

Thanks :)

File: data/sql/updates/pending_db_world/Roleplay.sql [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+DELETE FROM `creature_text` WHERE `CreatureID` = 27277;
+INSERT INTO `creature_text` (`CreatureID`, `GroupID`, `ID`, `Text`, `Type`, `Language`, `Probability`, `Emote`, `Duration`, `Sound`, `BroadcastTextId`, `TextRange`, `comment`) VALUES
+(27277, 0, 0, 'Alright Ben. We\'ve retrieved the amberseeds again. You know the drill.', 12, 0, 100, 0, 0, 0, 26350, 0, 'Master Woodsman Anderhol // Master Woodsman Anderhol');
+
+DELETE FROM `creature_text` WHERE `CreatureID` = 27071;
+INSERT INTO `creature_text` (`CreatureID`, `GroupID`, `ID`, `Text`, `Type`, `Language`, `Probability`, `Emote`, `Duration`, `Sound`, `BroadcastTextId`, `TextRange`, `comment`) VALUES
+(27071, 0, 0, 'I know, I know. Back in the bucket....', 12, 0, 100, 0, 0, 0, 26351, 0, 'Benjamin Jacobs // Benjamin Jacobs');
+
+DELETE FROM `waypoints` WHERE (entry = 27071) AND (pointid IN (1, 2));
+DELETE FROM `waypoints` WHERE (entry = 2707100) AND (pointid BETWEEN 1 AND 13);
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/Roleplay.sql:20]: backticks on the column names too

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**